src/Stratis.Bitcoin.Features.BlockStore/BlockRepository.cs

```diff
@@ -322,12 +337,53 @@ protected virtual void OnInsertTransactions(DBreeze.Transactions.Transaction dbr
             foreach ((Transaction transaction, Block block) in transactions)
             {
                 this.PerformanceCounter.AddRepositoryInsertCount(1);
-                dbreezeTransaction.Insert<byte[], uint256>("Transaction", transaction.GetHash().ToBytes(), block.GetHash());
+                dbreezeTransaction.Insert<byte[], uint256>(TransactionTableName, transaction.GetHash().ToBytes(), block.GetHash());
             }
 
             this.logger.LogTrace("(-)");
         }
 
+        /// <inheritdoc />
+        public Task ReIndexAsync()
+        {
+            this.logger.LogTrace("()");
+
+            Task task = Task.Run(() =>
+            {
+                this.logger.LogTrace("()");
+
+                using (DBreeze.Transactions.Transaction dbreezeTransaction = this.DBreeze.GetTransaction())
+                {
+                    if (this.TxIndex)
```

Comment:
I'm thinking of putting 
``` transaction.SynchronizeTables(BlockTableName, TransactionTableName);```

here, what do thing?

Reply:
Makes sense. This should only be called on node startup so it shouldn't be a problem, but of course we don't know what future developers will try to do, so I'll add it in.